src/model/adapter.py

```diff
@@ -120,41 +126,32 @@ def adapt_from_genotype(model, genotype_changes):
     # NOTES(Ali): review gene duplication logic below (`insert_feature` vs duplication check)
 
     operations = []
+    errors = []
 
-    to_remove = {}
-    to_add = {}
+    features_to_remove = {}
+    features_to_add = {}
 
     for change in full_genotype(genotype_changes).changes():
         if isinstance(change, gnomic.Mutation):
             old = change.old.features() if change.old else []
             for feature in old:
-                insert_feature(feature, to_remove, to_add)
+                insert_feature(feature, features_to_remove, features_to_add)
             new = change.new.features() if change.new else []
             for feature in new:
-                insert_feature(feature, to_add, to_remove)
+                insert_feature(feature, features_to_add, features_to_remove)
         if isinstance(change, gnomic.Plasmid):
             for feature in change.features():
-                insert_feature(feature, to_add, to_remove)
+                insert_feature(feature, features_to_add, features_to_remove)
 
     # Check if deletion-addition combination is in fact mutation and should be skipped.
-    # If the feature has DELETE_GENE suffix, consider this gene deleted
-    DELETE_GENE = 'delta8bp'  # the gene mutation which disables its functions
-    drop_remove, drop_add = [], []
-    for old_feature in to_remove:
-        for new_feature in to_add:
-            # NOTES(Ali): startswith?
+    for old_feature in features_to_remove:
```

Comment:
This design decision still escapes my comprehension. Why shouldn't I be allowed to be re-add something that was previously deleted? And if we **do** want this, what about the reverse case of deleting something that was previously added?